### PR TITLE
Some minor python tweaks

### PR DESCRIPTION
**Pull request by mattgodbolt**
A few things my IDE kept whinging about, and a little code duplication from my first proper look at the code.

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Very early sketches of thoughts here, Partouf; nothing I wouldn't easily leave how it was before - just things I couldn't help myself tweaking as I read through the code a bit.

File: bin/lib/library_builder.py [hunk 1]
```diff
@@ -586,26 +598,18 @@ def is_already_uploaded(self, buildfolder):
...
-                bininfo = BinaryInfo(self.logger, buildfolder, os.path.join(buildfolder, f'lib{lib}.a'))
-                libinfo = bininfo.cxx_info_from_binary()
-                archinfo = bininfo.arch_info_from_binary()
-                annotations['cxx11'] = libinfo['has_maybecxx11abi']
-                annotations['machine'] = archinfo['elf_machine']
-                annotations['osabi'] = archinfo['elf_osabi']
-
-        for lib in self.buildconfig.sharedliblink:
+        for lib in itertools.chain(self.buildconfig.staticliblink, self.buildconfig.sharedliblink):
+            # TODO - this is the same as the original code but I wonder if this needs to be *.so for shared?
```
Comment (partouf) [inline bin/lib/library_builder.py:612]: Oh, hmm, maybe yes.. haven't really tested shared libraries with this, so could very well be

**Review by partouf — APPROVED** [review]
Lgtm

**Comment by mattgodbolt** [comment]
Thanks Partouf :)

**Pull request merged**